lib/onebox/engine/amazon_onebox.rb

```diff
@@ -45,7 +45,11 @@ def http_params
       private
 
       def match
-        @match ||= @url.match(/(?:d|g)p\/(?:product\/|video\/detail\/)?(?<id>[^\/]+)(?:\/|$)/mi)
+        @match ||= @url.match(/(?:d|g)p\/(?:product\/|video\/detail\/)?(?<id>[A-Z0-9]+)(?:\/|\?|$)/mi)
```

Comment:
Is it intentional that some methods (`#match`, `#tld`) use `@url` variable and some (`#link`, `#data`) use `url` method?

Reply:
The pattern in Onebox seems to be that `@url` is the URL that Onebox was originally called with, and `url` within an engine is the 'resolved' URL that we're going to use to generate the Onebox content.